pkg/engine/update.go

```diff
@@ -657,9 +657,21 @@ func (acts *updateActions) OnResourceStepPost(
 }
 
 func (acts *updateActions) OnResourceOutputs(step deploy.Step) error {
-	acts.MapLock.Lock()
-	assertSeen(acts.Seen, step)
-	acts.MapLock.Unlock()
+	func() {
+		acts.MapLock.Lock()
+		defer acts.MapLock.Unlock()
+		assertSeen(acts.Seen, step)
+
+		var oldOutputs resource.PropertyMap
+		if step.Old() != nil {
+			oldOutputs = step.Old().Outputs
+		}
+		newOutputs := step.New().Outputs
+		diff := oldOutputs.Diff(newOutputs)
+		if diff.AnyChanges() {
+			acts.Ops[deploy.OpOutputChange] += len(diff.Adds) + len(diff.Deletes) + len(diff.Updates)
```

Comment:
Feels odd that this counts per-field changes rather than a count of how many resources made any change. Is there a reason to prefer this rather than just +1 here?

Reply:
In the scenarios where this number is shown to the user, this then means that the number shown matches the listed output diffs